conformance/generated/google/protobuf/any.luau

```diff
@@ -89,46 +102,99 @@ function _AnyImpl.decode(input: buffer): Any
 	return self
 end
 
-function _AnyImpl.jsonEncode(self: Any): any
-	local output: {
-		typeUrl: string?,
-		value: string?,
-	} = {}
+function _AnyImpl.jsonEncode(input: Any): { [string]: any }
+	local registry = typeRegistry.default
 
-	if self.type_url ~= nil and self.type_url ~= "" then
-		output.typeUrl = self.type_url
+	local unpacked = input:unpack(registry)
+	if unpacked == nil then
+		error("Cannot JSON-encode empty Any")
 	end
+	local json = unpacked:jsonEncode()
 
-	if self.value ~= nil and buffer.len(self.value) > 0 then
-		output.value = proto.json.serializeBuffer(self.value)
+	if typeof(json) == "string" then
+		return { ["@type"] = input.type_url, ["value"] = json }
 	end
 
-	return output
+	json["@type"] = input.type_url
+
+	return json
 end
 
 function _AnyImpl.jsonDecode(input: { [string]: any }): Any
-	local self = Any.new()
+	local registry = typeRegistry.default
 
-	if input.type_url ~= nil then
-		self.type_url = input.type_url
-	end
+	local typeUrl = input["@type"]
+	local typeName = typeUrl:sub(typeUrl:find("([^/]+)$"))
+	local payloadType = registry:findMessage(typeName)
 
-	if input.typeUrl ~= nil then
-		self.type_url = input.typeUrl
+	if payloadType == nil then
+		error("Could not find message type `" .. typeName .. "` for Any")
 	end
 
-	if input.value ~= nil then
-		self.value = proto.json.deserializeBuffer(input.value)
+	local typePrefix = typeUrl:sub(0, #typeUrl - #typeName - 1)
+
+	local json = table.clone(input)
+	-- For WKTs with special JSON representations, we need to decode input["value"] rather than
+	-- input.
+	local gpb = "google.protobuf."
+	if typeName:sub(0, #gpb) == gpb and input["value"] then
+		json = input["value"]
+	else
+		-- Clear this because it's not part of the JSON encoding of the payload itself.
+		json["@type"] = nil
 	end
 
-	return self
+	return _AnyImpl.pack(payloadType.jsonDecode(json), typePrefix)
 end
 
 _AnyImpl.descriptor = {
 	name = "Any",
 	fullName = "google.protobuf.Any",
 }
 
+function _AnyImpl.pack<T, F>(payload: proto.Message<T, F>, typePrefix: string?): Any
+	return Any.new({
+		type_url = (typePrefix or "apis.roblox.com/cloud") .. "/" .. payload.descriptor.fullName,
+		value = payload:encode(),
+	})
+end
+
+function _AnyImpl.unpack<T, F>(self: Any, registry: typeRegistry.TypeRegistry?): proto.Message<T, F>?
+	if self.value == nil then
+		return nil
+	end
+
+	if registry == nil then
+		registry = typeRegistry.default
+	end
+
+	local typeName = _AnyImpl.typeUrlToTypeName(self.type_url)
+	local payloadType = (registry :: typeRegistry.TypeRegistry):findMessage(typeName)
```

Comment:
Instead of this typecast, `assert` after line 169 with `assert(registry ~= nil, "Luau")`--it indicates that a better type solver would deem the assert unnecessary, but will nevertheless narrow the type.